src/cmn/ibprof_cmn.h

```diff
@@ -213,17 +246,22 @@ static INLINE int sys_dlcheck(const char *libname)
 {
 	int ret = IBPROF_ERR_NONE;
 	void *libc_handle = NULL;
+	const char *err = NULL;
 #if defined(__LINUX__)
 	dlerror();    /* Clear any existing error */
 	libc_handle = dlopen(libname, RTLD_LAZY);
 
-	if (!libc_handle || dlerror()) {
-		IBPROF_WARN("Can't find %s: %s\n", libname, dlerror());
+	err = dlerror();
+	if (!libc_handle || err) {
+		IBPROF_WARN("Can't find %s: %s\n", libname, err);
 		ret = IBPROF_ERR_NOT_EXIST;
 	}
 
-	if (libc_handle)
+	if (libc_handle) {
+                _libname = libname;
```

Comment:
how it will survive multi-threaded apps? the fact that static global var used to keep some state?